Modules/Relationships/src/Relationships.Infrastructure/Persistence/Database/Repository/RelationshipsRepository.cs

```diff
@@ -110,42 +123,50 @@ public async Task<DbPaginationResult<Relationship>> FindRelationshipsWithIds(IEn
     }
 
 
     public async Task Add(Relationship relationship, CancellationToken cancellationToken)
     {
         await _relationships.AddAsync(relationship, cancellationToken);
-        await _dbContext.SaveChangesAsync(cancellationToken);
+        await SaveContentOfLatestChange(relationship);
+        try
+        {
+            await _dbContext.SaveChangesAsync(cancellationToken);
+        }
+        catch (DbUpdateException ex)
+        {
+            if (ex.InnerException != null && ex.InnerException.Message.Contains(ConstraintNames.ONLY_ONE_ACTIVE_RELATIONSHIP_BETWEEN_TWO_IDENTITIES))
+                throw new OperationFailedException(ApplicationErrors.Relationship.RelationshipToTargetAlreadyExists(relationship.To));
+
+            throw;
+        }
     }
 
-    public async Task<bool> RelationshipBetweenTwoIdentitiesExists(IdentityAddress identityAddressA, IdentityAddress identityAddressB, CancellationToken cancellationToken)
+    public async Task<bool> RelationshipBetweenTwoIdentitiesExists(IdentityAddress identityAddressA,
+        IdentityAddress identityAddressB, CancellationToken cancellationToken)
     {
         return await _readOnlyRelationships
-                    .BetweenParticipants(identityAddressA, identityAddressB)
-                    .Where(r => r.Status != RelationshipStatus.Terminated && r.Status != RelationshipStatus.Rejected && r.Status != RelationshipStatus.Revoked)
-                    .AnyAsync(cancellationToken);
+            .BetweenParticipants(identityAddressA, identityAddressB)
+            .Where(r => r.Status != RelationshipStatus.Terminated && r.Status != RelationshipStatus.Rejected &&
+                        r.Status != RelationshipStatus.Revoked)
+            .AnyAsync(cancellationToken);
     }
 
-    public async Task SaveContentOfChangeRequest(RelationshipChangeRequest changeRequest)
+    public async Task SaveContentOfLatestChange(Relationship relationship)
     {
-        if (changeRequest.Content == null)
-            return;
+        var latestChange = relationship.Changes.OrderByDescending(c => c.CreatedAt).First();
 
-        _blobStorage.Add(_blobOptions.RootFolder, $"{changeRequest.Id}_Req", changeRequest.Content);
-        await _blobStorage.SaveAsync();
-    }
+        if(relationship.Status == RelationshipStatus.Pending && latestChange.Request.Content != null)
+            _blobStorage.Add(_blobOptions.RootFolder, $"{latestChange.Id}_Req", latestChange.Request.Content);
+        else if(latestChange.Response?.Content != null)
+            _blobStorage.Add(_blobOptions.RootFolder, $"{latestChange.Id}_Res", latestChange.Response.Content);
 
-    public async Task SaveContentOfChangeResponse(RelationshipChangeResponse changeResponse)
-    {
-        if (changeResponse.Content == null)
-            return;
-
-        _blobStorage.Add(_blobOptions.RootFolder, $"{changeResponse.Id}_Res", changeResponse.Content);
         await _blobStorage.SaveAsync();
```

Comment:
Are we no longer catching the exception?
```
try
{
    await _blobStorage.SaveAsync();
}
catch (BlobAlreadyExistsException)
{
    throw new DomainException(DomainErrors.ChangeRequestIsAlreadyCompleted(latestChange.Status));
}
```

Reply:
Shouldn't be necessary, and actually never was before, because the checks before should avoid this.

But I still catched it to be on the safe side.